NE-310 Enhancement proposal for HSTS route admission plugin (#849)
openshift/enhancements: enhancements/ingress/global-admission-hsts.md

#### Testing
Accompanying unit and e2e test changes will be added to exercise the `RequiredHSTSPolicies` type.
The admission plugin guarantees the following:

> * Routes with HSTS without a TLS termination policy of edge or re-encrypt may retain the annotation, but will not apply the header

Reviewer: It's HAProxy that would apply the header, so the language you have now in this item describes HAProxy behavior, not the admission plugin.  It should describe the plugin's behavior: 
```suggestion
* Routes with HSTS without a TLS termination policy of edge or re-encrypt will be accepted without checking the annotation.
```

Author: Thanks!
